src/reactThisBindingIssueRule.ts

```diff
@@ -243,8 +243,8 @@ class ReactThisBindingIssueRuleWalker extends ErrorTolerantWalker {
 
                                         const rightPropText = AstUtils.getFunctionTarget(callExpression);
                                         if (leftPropText === rightPropText) {
-                                            if (result.indexOf(rightPropText) === -1) {
-                                                result.push(rightPropText);
+                                            if (!result.has(rightPropText)) {
```

Comment:
Fun fact: you can actually just skip this `.has`, since a `.add` on something that already exists in the Set does nothing.